### DISCUSSION TITLE
Trying out the hydrate function and getting TypeError

### DISCUSSION THREAD

**Discussion by mirabledictu** [solidjs/solid#599] [Q&A]
Hi, I'm trying the [hydrate](https://www.solidjs.com/docs/latest/api#hydrate) function with a fresh solid starter for research purposes with Vite and I'm getting this type error:

`TypeError: Cannot read property 'resources' of undefined` 

It's in this part

```ts
function hydrate(code, element) {
  sharedConfig.resources = globalThis._$HYDRATION.resources; // this one
  sharedConfig.completed = globalThis._$HYDRATION.completed;
  sharedConfig.events = globalThis._$HYDRATION.events;
  sharedConfig.context = {
    id: "",
    count: 0,
    loadResource: globalThis._$HYDRATION.loadResource
  };
  sharedConfig.registry = new Map();
  gatherHydratable(element);
  const dispose = render(code, element, [...element.childNodes]);
  sharedConfig.context = null;
  return dispose;
}
```

Code:

```tsx
import { hydrate } from "solid-js/web";
import App from "./App";
hydrate(() => <App />, document.getElementById("root"));
```

Am I missing something?

**Comment by ryansolid** [comment]
I suspect the hydration script isn't on the page. We have a small bootloader of sorts that lets us do a bunch of hydration-related tasks before even the core Solid JavaScript library is loaded. We capture events for playback and set code for handling async resources if they load before the main JS bundle does (this can happen when streaming). 

You want to server render the hydration script into your page. https://www.solidjs.com/guide#hydration-script
